src/ToricVarieties/NormalToricVarieties/attributes.jl

```diff
@@ -9,7 +9,9 @@
 Computes the dimension of the normal toric variety `v`.
 """
 function dim(v::AbstractNormalToricVariety)
-    return pm_object(v).FAN_DIM::Int
+    return get_attribute!(v, :dim) do
+        return pm_object(v).FAN_DIM::Int
+    end
```

Comment:
I think some care is needed to retain type stability; the type assertion needs to be moved to the outer return. I think this might work (but didn't test):
```suggestion
    return get_attribute!(v, :dim)::Int do
        return pm_object(v).FAN_DIM
    end
```
You can verify if it worked by (a) calling it (to makes sure I got the syntax right, I am not sure), and (b) using `@code_warntype` on it to verify the return type is inferred correctly.

Reply:
Thank you for the suggestion @fingolfin . For me, this change leads to the following:

> ERROR: LoadError: LoadError: LoadError: syntax: "function" at Oscar.jl/src/ToricVarieties/NormalToricVarieties/attributes.jl:11 expected "end", got "do"

Reply:
For all it matters, here is a MWE via `@code_warntype`, which supports your suspicion that type stability requires more care:

> v = toric_projective_space(2)
A normal, non-affine, smooth, projective, gorenstein, q-gorenstein, fano, 2-dimensional toric variety without torusfactor

> julia> @code_warntype dim(v)
Variables
  #self#::Core.Const(AbstractAlgebra.Generic.dim)
  v::NormalToricVariety
  #1418::Oscar.var"#1418#1419"{NormalToricVariety}

Body::Any
1 ─ %1 = Oscar.:(var"#1418#1419")::Core.Const(Oscar.var"#1418#1419")
│   %2 = Core.typeof(v)::Core.Const(NormalToricVariety)
│   %3 = Core.apply_type(%1, %2)::Core.Const(Oscar.var"#1418#1419"{NormalToricVariety})
│        (#1418 = %new(%3, v))
│   %5 = #1418::Oscar.var"#1418#1419"{NormalToricVariety}
│   %6 = Oscar.get_attribute!(%5, v, :dim)::Any
└──      return %6

Hence, if I read this information correctly, `Body::Any` show that the return type is `Any` rather than `Int`.

Reply:
How about turning function `dim(v::AbstractNormalToricVariety)` into `dim(v::AbstractNormalToricVariety)::Int`. This would appear to work and - base on @code_warntype - enusure that the return value is always `Int64`.

Reply:
I am uncertain if the failures in the runs of the nightly build (ERROR: LoadError: Failed to precompile CxxWrap) are correlated to the changes in this PR.

Reply:
writing `dim(v::...)::Int` has different meaning, though: it *converts* the result to `Int`, by inserting a call to `convert(Int, ...)`; while a type assertion does just that, it checks that the value is an `Int` (if not, an exception is thrown).

Anyway, let's not worry about this for the moment. I describe a potential future solution for this in another comment.